src/services/document-service.js

```diff
@@ -461,6 +474,67 @@ class DocumentService {
     }
   }
 
+  async regenerateDocument(documentKey) {
+    let lock;
+
+    try {
+      lock = await this.documentLockStore.takeLock(documentKey);
+
+      await this.transactionRunner.run(async session => {
+        const existingDocumentRevisions = await this._getAllDocumentRevisionsByKey(documentKey, session);
+
+        const document = this._buildDocumentFromRevisions(existingDocumentRevisions);
+
+        logger.info(`Saving document '${documentKey}' with revision ${document.revision}`);
+        await this.documentStore.save(document, { session });
+      });
+    } finally {
+      if (lock) {
+        await this.documentLockStore.releaseLock(lock);
+      }
+    }
+  }
+
+  async createRegenerateDocumentsBatch(user) {
+    const existingActiveBatch = await this.batchStore.findOne({
+      batchType: BATCH_TYPE.regenerateDocuments,
```

Comment:
Same here, `documentsRegeneration` ?
And we could align the existing one too, cause I think a better name would have been `documentsImport` as a batch name/type.

Reply:
Maybe rather `documentRegeneration` and `documentImport` as we have a `document-import-task-processor.js` and soon a `document-regeneration-task-processor.js`. In these kind of compound words we always use the singular for the left part (cf. `documents` vs `documentService`), except when it's a fixed name

Reply:
But be aware that we use these constants in the database!

Reply:
I think we need to clear out the dbs for OMA so this is the best time to do this. What do you think? (we can skip having a migration since it's only test data for now.

Reply:
I've done the "documentRegeneration" part of this comment, shall I do the second part now or add it to the story about unifying the batches? 
I'm a bit afraid this PR will touch quite a few files if I do "documentImport" now, so I think in the end I'd opt for the latter.

Reply:
Done, with migration